### PR TITLE
make a model for room creation

### PR DESCRIPTION
**Pull request by 4nkitd**
image

### CODE REVIEW

File: src/components/rooms/Rooms.tsx [hunk 1]
```diff
@@ -289,29 +327,37 @@ export const RoomCreationForm: FC = () => {
...
-
     if (loading) {
         return (<p>...Loading</p>);
     } else if (error) {
         return (<p>...Mutation Failed</p>);
     }
 
+    const roomForm = <FormWrapper>
+        <FormHeader>
+            <Title>New room</Title>
```
Comment (chalkedgoose) [inline src/components/rooms/Rooms.tsx:338]: Can we make this into a proper arrow or function component?

File: src/components/rooms/Rooms.tsx [hunk 2]
```diff
@@ -289,29 +327,37 @@ export const RoomCreationForm: FC = () => {
...
                     {data ? (<div>
                         <p>Room {data.createRoom.id}: {data.createRoom.name} created</p>
-                    </div>) : (<form onSubmit={handleSubmit(onSubmit)}>
-                        <input placeholder="Room Name" {...register("name")} />
-                        <p>{errors.name?.message}</p>
-                        <input placeholder="Room Description" {...register("description")} />
-                        <p>{errors.description?.message}</p>
-                        <input type="Submit" />
-                    </form>)}
+                    </div>) : (roomForm)}
```
Comment (chalkedgoose) [inline src/components/rooms/Rooms.tsx:358]: Should be called like <RoomForm/>

**Review by lucemans — APPROVED** [review]

File: src/components/rooms/Rooms.tsx [hunk 3]
```diff
@@ -289,29 +324,46 @@ export const RoomCreationForm: FC = () => {
...
                         <p>Room {data.createRoom.id}: {data.createRoom.name} created</p>
-                    </div>) : (<form onSubmit={handleSubmit(onSubmit)}>
-                        <input placeholder="Room Name" {...register("name")} />
-                        <p>{errors.name?.message}</p>
-                        <input placeholder="Room Description" {...register("description")} />
-                        <p>{errors.description?.message}</p>
-                        <input type="Submit" />
-                    </form>)}
+                        <CloseButton onClick={toggleExpanded} > X </CloseButton>
+                    </Row>) : (roomForm())}
```
Comment (chalkedgoose) [inline src/components/rooms/Rooms.tsx:364]: Change to Proper JSX Element

**Review by chalkedgoose — CHANGES_REQUESTED** [review]

File: src/components/rooms/Rooms.tsx (hunk 3 above)
Comment (chalkedgoose) [reply src/components/rooms/Rooms.tsx:364]: <RoomForm/>

Comment (chalkedgoose) [reply src/components/rooms/Rooms.tsx:364]: <RoomForm/>

Comment (chalkedgoose) [reply src/components/rooms/Rooms.tsx]: `
<RoomForm/>
`

**Review by chalkedgoose — APPROVED** [review]

**Pull request merged**